api/src/main/java/org/apache/iceberg/Files.java

```diff
@@ -60,9 +60,8 @@ public PositionOutputStream create() {
 
       if (!file.getParentFile().isDirectory() && !file.getParentFile().mkdirs()) {
         throw new RuntimeIOException(
-            String.format(
                 "Failed to create the file's directory at %s.",
-                file.getParentFile().getAbsolutePath()));
+                file.getParentFile().getAbsolutePath());
```

Comment:
I personally feel that replacing deprecated exceptions is out of scope for this PR. I think it would be cleaner / easier if the deprecated exception was replaced in another PR in one go or just left as is for now. As it stands, the PR is already 71 files changed.

Reply:
Separate PR is good.

Reply:
We're also keeping it for a release so that people have a chance to update any code that catches it.